python/orion_finance_sdk_py/contracts.py

```
@@ -804,15 +972,168 @@ def share_price(self) -> int:
+
+        # Normalize price keys for lookup (checksum + lowercase)
+        price_by_lower = {addr.lower(): price for addr, price in prices.items()}
+
+        values: dict[str, int] = {}
+        for token, shares in portfolio.items():
+            checksum = Web3.to_checksum_address(token)
+            if checksum.lower() not in price_by_lower:
+                raise ValueError(
+                    f"No PIT price for portfolio token {checksum}. "
+                    "Token may not be in the investment universe."
+                )
```

Ruff (0.15.21)

[warning] Avoid specifying long messages outside the exception class

(TRY003)